src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -614,23 +558,85 @@
    jruby-service :- (schema/protocol jruby-protocol/JRubyPuppetService)]
   (fn [request]
     (let [environment (jruby-request/get-environment-from-request request)
+          svc-key (info-service request)
           request-tag (if-none-match-from-request request)]
       (if (and request-tag
                (= request-tag
-                  (jruby-protocol/get-environment-class-info-tag
+                  (jruby-protocol/get-environment-info-tag
                    jruby-service
-                   environment)))
+                   environment
+                   svc-key)))
         (not-modified-response request-tag)
         (handler request)))))
 
 (schema/defn ^:always-validate
-  environment-class-handler :- IFn
-  "Handler for processing an incoming environment_classes Ring request"
-  [jruby-service :- (schema/protocol jruby-protocol/JRubyPuppetService)
-   environment-class-cache-enabled :- schema/Bool]
-  (->
-   (environment-class-info-fn jruby-service
-                              environment-class-cache-enabled)
+  raw-transports->response-map
+  [data :- List
+   env :- schema/Str]
+  (sort-nested-info-maps
+    {:name env
+     :transports data}))
+
+(schema/defn ^:always-validate
+  check-cache! :- ringutils/RingResponse
```

Comment:
should be named closer to what it does, i'd expect `check-cache` to be used to check the cache early in the request to avoid doing work, but this is used after only for updating the cache